uint32 vhcid
uint32 cmd
